### PR TITLE
Implement new product naming scheme

### PR DESCRIPTION
**Pull request by asjohnston-asf**

### CODE REVIEW

File: hyp3_rtc_gamma/__main__.py [hunk 1]
```diff
@@ -214,6 +212,16 @@ def find_product(dir_):
...
 
 
+def find_product_name(dir_):
+    files = glob.glob(f'{dir_}/**')
+    readme_file = None
+    for file in files:
+        if file.endswith('.README.txt'):
+            readme_file = file
+            break
+    return readme_file.split('.')[0]
```
Comment (asjohnston-asf) [inline hyp3_rtc_gamma/__main__.py:222]: This implementation is pretty lazy and doesn't have tests.  Could refactor all 5 `find_...` functions, they're mostly copy/paste of each other, but I'm not feeling up to it right now.

File: hyp3_rtc_gamma/rtc_sentinel.py [hunk 2]
```diff
@@ -38,15 +39,28 @@
...
-    g = 'g' if gamma0 else 's'
+
+    if orbit_file is None:
+        o = 'O'
+    elif 'POEORB' in orbit_file:
+        o = 'P'
+    elif 'RESORB' in orbit_file:
+        o = 'R'
+    else:
+        o = 'O'
```
Comment (asjohnston-asf) [inline hyp3_rtc_gamma/rtc_sentinel.py:55]: Open to suggestions here, not in love with this implementation.

File: hyp3_rtc_gamma/rtc_sentinel.py [hunk 3]
```diff
@@ -733,7 +747,7 @@ def rtc_sentinel_gamma(in_file,
     create_consolidated_log(out_name, lo_flag, dead_flag, match_flag, gamma_flag, roi,
                             shape, pwr_flag, filter_flag, pol, looks, log_file, smooth, terms,
                             no_cross_pol, par)
-    return 'PRODUCT'
+    return 'PRODUCT', out_name
```
Comment (asjohnston-asf) [inline hyp3_rtc_gamma/rtc_sentinel.py:750]: Both v1 and v2 entrypoints rename the PRODUCT dir to the product name.  I'd like to see that happen here and just return the one value, but I don't feel like digging through the v1 post-processing code to see if anything is hard-coded to look for PRODUCT/.

File: hyp3_rtc_gamma/etc/README_RTC_GAMMA.txt [hunk 4]
```diff
@@ -5,20 +5,24 @@ This folder contains radiometric terrain corrected (RTC) products and their asso
...
-The source granule used to generate the products contained in this folder is:  
-[GRAN_NAME]  
+The folder and each of its contents all share the same base name, using the following convention:
+S1x_yy_aaaaaaaaTbbbbbb_ppo_RTCzz_G_defkl_ssssss
+x:          Sentinel-1 Mission (A or B)
+yy:         Beam Mode
+aaaaaaaa:   Start Date of Acquisition (YYYYMMDD)
+bbbbbb:     Start Time of Acquisition (HHMMSS)
+pp:         Polarization
+o:          Orbit Type (Precise, Restituted, or Predicted)
```
Comment (asjohnston-asf) [inline hyp3_rtc_gamma/etc/README_RTC_GAMMA.txt:15]: I don't like using `O` for `Predicted`, they don't feel connected to me at all, I'll follow up with Tom.

File: hyp3_rtc_gamma/rtc_sentinel.py (hunk 2 above)
Comment (Jlrine2) [reply hyp3_rtc_gamma/rtc_sentinel.py:55]: I agree, but nothing much better comes to mind
```python
replacements = {
    'POEORB': 'P',
    'RESORB': 'R',
    'default': 'O'
}
if orbit_file is not None:
    o = [v for k, v in replacements if k in orbit_file][0]
if o is not None:
    o = 'O'
```
for example is much worse

File: hyp3_rtc_gamma/rtc_sentinel.py (hunk 3 above)
Comment (Jlrine2) [reply hyp3_rtc_gamma/rtc_sentinel.py:750]: I think we are making a clear improvement, might be good to push on and document these wants

File: hyp3_rtc_gamma/__main__.py (hunk 1 above)
Comment (jhkennedy) [reply hyp3_rtc_gamma/__main__.py:222]: We'll do some cleanup and moving stuff into hyp3lib as we start adding more processes, so we can circle back to this then

File: hyp3_rtc_gamma/__main__.py [hunk 5]
```diff
@@ -346,20 +352,18 @@ def process_rtc_gamma(cfg, n):
...
-                find_and_remove(product, '_dem.tif')
+                find_and_remove(product_dir, '_dem.tif')
             if extra_arg_is(cfg, "include_inc", "no"):
-                find_and_remove(product, '_inc_map.tif')
+                find_and_remove(product_dir, '_inc_map.tif')
 
-            out_name = build_output_name(g, cfg['workdir'], opts_str + cfg['suffix'])
-            if "STD_L0_F" in out_name:
-                out_name = out_name.replace("_L0", "")
+            out_name = find_product_name(product_dir)
```
Comment (asjohnston-asf) [inline hyp3_rtc_gamma/__main__.py:366]: need to address non-sentinel products, I think the other use case is PALSAR via rtc_legacy.py

Comment (jhkennedy) [reply hyp3_rtc_gamma/__main__.py:344]: I think other products will need to be addressed separately; there is a bit more work to do from what I can tell to get them supported in the containers (missing dependencies). Maybe just make an issue for this

Comment (asjohnston-asf) [reply hyp3_rtc_gamma/__main__.py:344]: Yeah, should just get confirmation from Kirk that things are already broken in production, in which case this change won't make things any worse.

**Comment by asjohnston-asf** [comment]
container `2.0.7.dev20_g5992e11` passes a happy path test for v1 against the hyp3-test database

**Comment by asjohnston-asf** [comment]
found a pathing problem applying slc orbit files, get_orb downloads to the root working directory, but applying the orbit file happens in a subdirectory at https://github.com/ASFHyP3/hyp3-lib/blob/develop/hyp3lib/par_s1_slc_single.py#L68 , so the orbit file isn't found

File: hyp3_rtc_gamma/rtc_sentinel.py [hunk 6]
```diff
@@ -657,6 +657,8 @@ def rtc_sentinel_gamma(in_file,
     report_kwargs(in_file, out_name, res, dem, roi, shape, match_flag, dead_flag, gamma_flag, lo_flag,
                   pwr_flag, filter_flag, looks, terms, par, no_cross_pol, smooth, area, orbit_file)
 
+    orbit_file = os.path.join(os.getcwd(), orbit_file)  # ingest_S1_granule requires absolute path
```
Comment (jhkennedy) [inline hyp3_rtc_gamma/rtc_sentinel.py:660]: might be more explicit to do an `os.path.abspath`

Comment (jhkennedy) [reply hyp3_rtc_gamma/rtc_sentinel.py:660]: and could probably me moved into your helper `fetch_orbit_file` function

File: hyp3_rtc_gamma/etc/README_RTC_GAMMA.txt (hunk 4 above)
Comment (jhkennedy) [reply hyp3_rtc_gamma/etc/README_RTC_GAMMA.txt:15]: Sounds like, U for Unknown would be a better letter as we're not downloading the Predicted files (unless that's also included in the SAFE and we're pulling it from there)

File: hyp3_rtc_gamma/rtc_sentinel.py (hunk 6 above)
Comment (asjohnston-asf) [reply hyp3_rtc_gamma/rtc_sentinel.py:660]: I prefer handing just the filename to `get_product_name` and `report_kwargs` (for now).  Implemented `os.path.abspath()`.

**Pull request merged**